packages/core/src/browser/messaging/ws-connection-provider.ts

```diff
@@ -45,7 +51,11 @@ export class WebSocketConnectionProvider extends AbstractConnectionProvider<WebS
         const url = this.createWebSocketUrl(WebSocketChannel.wsPath);
         const socket = this.createWebSocket(url);
         socket.onerror = console.error;
+        socket.onopen = () => {
+            this.onSocketOpenedEmitter.fire(undefined);
```

Comment:
Please move the calls to protected methods. So one can override and customize.

Reply:
Fixed. There are two new protected methods which are fire the events.